applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1225,100 +1163,80 @@ static struct bt_bap_stream_ops stream_ops = {
-			work_data.channel_index, ret, work_data.retries);
-		LOG_DBG("k_msg_put: ch: %d, dir: %d, retries %d", work_data.channel_index,
-			work_data.dir, work_data.retries);
-		ret = k_msgq_put(&kwork_msgq, &work_data, K_NO_WAIT);
-		if (ret) {
-			LOG_ERR("No space in the queue for work_data");
-			return;
-		}
-		/* Delay added to prevent controller overloading */
-		if (work_data.dir == BT_AUDIO_DIR_SINK) {
-			k_work_reschedule(&headsets[work_data.channel_index].stream_start_sink_work,
-					  K_MSEC(CIS_CONN_RETRY_DELAY_MS));
-		} else if (work_data.dir == BT_AUDIO_DIR_SOURCE) {
-			k_work_reschedule(
-				&headsets[work_data.channel_index].stream_start_source_work,
-				K_MSEC(CIS_CONN_RETRY_DELAY_MS));
-		}
-	} else if (ret != 0) {
-		LOG_WRN("Failed to establish CIS, ret = %d", ret);
-		/** The connection could have invalid configs, or abnormal behavior cause
-		 * the CIS failed to establish. Sending an event for triggering
-		 * disconnection could clean up the abnormal state and restart the
-		 * connection.
-		 */
-		le_audio_event_publish(LE_AUDIO_EVT_NO_VALID_CFG,
-				       headsets[work_data.channel_index].headset_conn,
-				       work_data.dir);
-	} else if (k_msgq_peek(&kwork_msgq, &work_data) != -ENOMSG) {
-		if (work_data.dir == BT_AUDIO_DIR_SINK &&
-		    !k_work_delayable_is_pending(
-			    &headsets[work_data.channel_index].stream_start_sink_work)) {
-			k_work_reschedule(&headsets[work_data.channel_index].stream_start_sink_work,
-					  K_MSEC(CIS_CONN_RETRY_DELAY_MS));
-		} else if (work_data.dir == BT_AUDIO_DIR_SOURCE &&
-			   !k_work_delayable_is_pending(
-				   &headsets[work_data.channel_index].stream_start_source_work)) {
-			k_work_reschedule(
-				&headsets[work_data.channel_index].stream_start_source_work,
-				K_MSEC(CIS_CONN_RETRY_DELAY_MS));
-		}
-	}
-}
+static struct bt_bap_unicast_client_cb unicast_client_cbs = {
+	.location = unicast_client_location_cb,
+	.available_contexts = available_contexts_cb,
+	.pac_record = pac_record_cb,
+	.endpoint = endpoint_cb,
+	.discover = discover_cb,
+};
 
 static void disconnected_headset_cleanup(uint8_t chan_idx)
 {
 	headsets[chan_idx].headset_conn = NULL;
-	k_work_cancel_delayable(&headsets[chan_idx].stream_start_sink_work);
 	headsets[chan_idx].sink_ep = NULL;
 	memset(headsets[chan_idx].sink_codec_cap, 0, sizeof(headsets[chan_idx].sink_codec_cap));
-	k_work_cancel_delayable(&headsets[chan_idx].stream_start_source_work);
 	headsets[chan_idx].source_ep = NULL;
 	memset(headsets[chan_idx].source_codec_cap, 0, sizeof(headsets[chan_idx].source_codec_cap));
 
 	headsets[chan_idx].num_sink_eps = 0;
 	headsets[chan_idx].num_source_eps = 0;
 }
 
+static void unicast_discovery_complete_cb(struct bt_conn *conn, int err,
+					  const struct bt_csip_set_coordinator_csis_inst *csis_inst)
+{
+
+	LOG_DBG("Unicast discovery complete cb");
+	LOG_DBG("\tErr: %d, set_size: %d, key: %s", err, csis_inst->info.set_size,
```

Comment:
Suggest that for all these callbacks, we do a check if there is an err, and if so, we print with an elevated warning level.

Reply:
Nice, but should we `return` if there is an error?

Reply:
No need, the code that follows is not related to the conn that failed